corehq/apps/app_manager/static/app_manager/js/preview_app.js

```diff
@@ -110,7 +119,18 @@ hqDefine('app_manager/js/preview_app.js', function() {
         };
         $(window).on(module.EVENTS.RESIZE, _resizeAppPreview);
         layoutController.utils.setBalancePreviewFn(_resizeAppPreview);
-        $('.js-preview-back').click(_private.navigateBack);
+        $('.js-preview-back').click(_private.triggerPreviewEvent.bind(this, 'back'));
+        $('.js-preview-refresh').click(function() {
+            $(module.SELECTORS.BTN_REFRESH).removeClass('app-out-of-date');
+            _private.triggerPreviewEvent('refresh');
+        });
+        $(document).ajaxComplete(function(e, xhr, options) {
+            if (/edit_form_attr/.test(options.url) ||
```

Comment:
ok we'll probably always have a refresh button (did some talking with amelia) but this is still the least invasive approach. otherwise we'd to either inject some JS in every button that updated the app or continually poll HQ to see if the latest build has been updated